lib/spack/spack/solver/concretize.lp

```diff
@@ -1245,7 +1245,9 @@ error(100, "Compiler {1}@{2} requested for {0} cannot be found. Set install_miss
 % propagate flags when compiler match
 can_inherit_flags(PackageNode, DependencyNode, FlagType)
   :- same_compiler(PackageNode, DependencyNode),
-     not attr("node_flag_set", DependencyNode, FlagType, _),
+     % Disable to pass propagate_and_spec_dep, propagate_and_pkg_dep
+     % and propagate_and_require
+     % not attr("node_flag_set", DependencyNode, FlagType, _, _),
```

Comment:
This follow-up to https://github.com/spack/spack/pull/41049/files#r1591609679 is required to make all `test_propagate_and_*` tests pass except for `propagate_and_compiler_cfg`. In that sense, it isn't needed for https://github.com/spack/spack/pull/41049#issuecomment-2096014945, but I think this relaxation is sensible (I ran all the tests that made sense to me locally and am thinking the CI full run will confirm whether there are any flaws with this).

Reply:
I don't have opinions on this one, but I think it changes previous semantic. Before, setting a flag, would stop propagation through that node.

Reply:
> Before, setting a flag,

By "before" I don't think you mean before this comment: but rather commits that were made by you after this comment, but that landed before this PR merged. More discussion at https://github.com/spack/spack/pull/41049#discussion_r1674822909 - we did not enforce this until after May 6 2024

Reply:
Resolved in https://github.com/spack/spack/pull/41049#discussion_r1732076592